WcaOnRails/app/webpacker/components/Competitions/MyCompetitions/index.jsx

```diff
@@ -0,0 +1,310 @@
+    statusText += I18n.t('competitions.messages.tooltip_deleted');
+  }
+  if (competition['confirmed?']) {
+    statusText += I18n.t('competitions.messages.confirmed_visible');
+  } else if (competition['visible?']) {
+    statusText += I18n.t('competitions.messages.confirmed_not_visible');
+  } else {
+    statusText += I18n.t('competitions.messages.not_confirmed_not_visible');
+  }
+  return statusText;
+};
+
+const competitionStatusIcon = (competition) => {
+  if (competition['registration_not_yet_opened?']) {
+    return <Icon name="clock" color="blue" />;
+  }
+  if (competition['registration_past?']) {
+    return <Icon name="user times" color="red" />;
+  }
+  if (competition['registration_full?']) {
+    return <Icon name="user clock" color="orange" />;
+  }
+  return <Icon name="user plus" color="green" />;
+};
+
+function ReportTableCell({ permissions, competitionId, isReportPosted }) {
+  return (
+    <Table.Cell>
+      {(permissions.can_administer_competitions.scope === '*' || permissions.can_administer_competitions.scope.includes(competitionId)) && (
+      <>
+        <Popup
+          content={I18n.t('competitions.my_competitions_table.report')}
+          trigger={(
+            <a href={competitionReportUrl(competitionId)}>
+              <Icon name="file alternate" />
+            </a>
+          )}
+        />
+        <Popup
+          content={I18n.t('competitions.my_competitions_table.edit_report')}
+          trigger={(
+            <a href={competitionReportEditUrl(competitionId)}>
+              <Icon name="edit" />
+            </a>
+          )}
+        />
+        { !isReportPosted
+          && permissions.can_administer_competitions.scope.includes(competitionId) && (
+          <Popup
+            content={I18n.t('competitions.my_competitions_table.missing_report')}
+            trigger={(
+              <Icon name="warning" />
+            )}
+          />
+        )}
+      </>
+      )}
+    </Table.Cell>
+  );
+}
+
+function UpcomingCompetitionTable({
+  competitions, permissions, registrationStatuses, shouldShowRegistrationStatus = false,
+}) {
+  return (
+    <Table>
+      <TableHeader>
+        <Table.Row>
+          { shouldShowRegistrationStatus && <Table.HeaderCell /> }
+          <Table.HeaderCell>
+            {I18n.t('competitions.adjacent_competitions.name')}
+          </Table.HeaderCell>
+          <Table.HeaderCell>
+            {I18n.t('competitions.adjacent_competitions.location')}
+          </Table.HeaderCell>
+          <Table.HeaderCell>
+            {I18n.t('competitions.adjacent_competitions.date')}
+          </Table.HeaderCell>
+          <Table.HeaderCell />
+          <Table.HeaderCell />
+          <Table.HeaderCell />
+          <Table.HeaderCell />
+          <Table.HeaderCell />
+        </Table.Row>
+      </TableHeader>
+
+      <TableBody>
+        {competitions.map((competition) => (
+          <Popup
+            position="top center"
```

Comment:
We should really have lint errors turned on for react.

Don't forget to be adding `key` to the top most react elements in a map. Table.Row doesn't need it, popup should need it.